src/main/java/com/auth0/IdTokenVerifier.java

```diff
@@ -55,6 +55,17 @@ void verify(String token, Options verifyOptions) throws TokenValidationException
             throw new TokenValidationException(String.format("Audience (aud) claim mismatch in the ID token; expected \"%s\" but found \"%s\"", verifyOptions.audience, decoded.getAudience()));
         }
 
+        // validate org if set
+        if (verifyOptions.organization != null) {
+            String orgIdClaim = decoded.getClaim("org_id").asString();
+            if (isEmpty(orgIdClaim)) {
+                throw new TokenValidationException("Organization Id (org_id) claim must be a string present in the ID token");
+            }
+            if (!verifyOptions.organization.equals(orgIdClaim)) {
+                throw new TokenValidationException(String.format("Organization (org_id) claim mismatch in the ID token; expected \"%s\" but found \"%s\"", verifyOptions.organization, orgIdClaim));
+            }
+        }
+
```

Comment:
This is duplicated logic from ID Token validation in `auth0-java`. We have work defined to refactor this library to use the ID Token validation in `auth0-java`, but it's likely something that would have to be done in a new major, due to ensuring backwards-compatibility within this release stream. I'm not _entirely_ convinced it can't be done in a minor release, but it would require some deeper investigation that goes beyond this change. SDK-1637 exists for that work.